spark/src/main/java/org/apache/iceberg/spark/actions/BaseRewriteManifestsSparkAction.java

```diff
@@ -86,26 +91,29 @@
   private static final String USE_CACHING = "use-caching";
   private static final boolean USE_CACHING_DEFAULT = true;
 
-  private final Encoder<ManifestFile> manifestEncoder;
+  private final Encoder<RepairManifestHelper.RepairedManifestFile> manifestEncoder;
   private final Table table;
   private final int formatVersion;
   private final FileIO fileIO;
+  private final SerializableConfiguration hadoopConf;
   private final long targetManifestSizeBytes;
 
   private PartitionSpec spec = null;
   private Predicate<ManifestFile> predicate = manifest -> true;
   private String stagingLocation = null;
+  private RepairMode mode = RepairMode.NONE;
 
   public BaseRewriteManifestsSparkAction(SparkSession spark, Table table) {
     super(spark);
-    this.manifestEncoder = Encoders.javaSerialization(ManifestFile.class);
+    this.manifestEncoder = Encoders.javaSerialization(RepairManifestHelper.RepairedManifestFile.class);
```

Comment:
Are these changes actually needed? The `RepairedManifestFile` interface sends back the fields that were repaired. Is that specific to a file and empty if, for example, the length for all manifest entries were already correct?

I don't see much benefit to doing it that way. If we were to have more specific repair operations, then we don't need that interface at all because we'd already know what manifests fields are being fixed.

Reply:
OK yes I was trying to think something useful to return to user.  But maybe its not very useful as its not specific as per discussion with @flyrain  earlier in the review, and returning all patched manifest-entries is a bit overkill.  I'm ok for just return the list of repaired ManifestFile